src/panels/calendar/ha-panel-calendar.js

```diff
@@ -0,0 +1,233 @@
+                    <span class="calendar_color_spacer"></span>
+                    [[item.name]]
+                  </paper-item>
+                </template>
+              </paper-listbox>
+            </paper-card>
+          </div>
+          <div class="flex layout horizontal wrap">
+            <ha-big-calendar
+              default-date="[[currentDate]]"
+              default-view="[[currentView]]"
+              view-updated="[[viewUpdated()]]"
+              date-updated="[[dateUpdated()]]"
+              events="[[events]]">
+            </ha-big-calendar>
+          </div>
+        </div>
+      </app-header-layout>`;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._fetchData = this._fetchData.bind(this);
+    // TODO implement calendar_updated event
+    // this.hass.connection.subscribeEvents(this._fetchData, 'calendar_updated')
+    //  .then(function (unsub) { this._unsubEvents = unsub; }.bind(this));
+    this._fetchCalendar();
+  }
+
+  _fetchCalendar() {
+    // Fetch calendar list
+    this.hass.callApi('get', 'calendars')
+      .then((result) => {
+        this.calendars = result;
+      });
+  }
+
+  _fetchData() {
+    var dates = this._getDateRange();
+    var start = dates[0];
+    var end = dates[1];
+    // Fetch calendar list
+    this._fetchCalendar();
+    // Fetch events for selected calendar
+    const params = encodeURI(`?start=${start}&end=${end}`);
+    const calls = this.selectedCalendars.map(cal => this.hass.callApi('get', `calendar/${cal}${params}`));
+    Promise.all(calls).then((results) => {
+      var tmpEvents = [];
+
+      results.map((res) => {
+        res.map((ev) => {
+          ev.start = new Date(ev.start);
+          if (ev.end) {
+            ev.end = new Date(ev.end);
+          } else {
+            ev.end = null;
+          }
+          tmpEvents.push(ev);
+          return null;
+        });
+        return null;
+      });
+      this.events = tmpEvents;
+    });
+  }
+
+  _getDateRange() {
+    var startDate;
+    var endDate;
+    if (this.currentView === 'day') {
+      startDate = moment(this.currentDate).startOf('day');
+      endDate = moment(this.currentDate).startOf('day');
+    } else if (this.currentView === 'isoWeek') {
+      startDate = moment(this.currentDate).startOf('isoWeek');
+      endDate = moment(this.currentDate).endOf('isoWeek');
+    } else if (this.currentView === 'month') {
+      startDate = moment(this.currentDate).startOf('month').subtract(7, 'days');
+      endDate = moment(this.currentDate).endOf('month').add(7, 'days');
+    } else if (this.currentView === 'agenda') {
+      startDate = moment(this.currentDate).startOf('day');
+      endDate = moment(this.currentDate).endOf('day').add(1, 'month');
+    }
+    return [startDate.toISOString(), endDate.toISOString()];
+  }
+
+  viewUpdated() {
+    // Calendar view changed
+    var pol = this;
+    return function (viewName) {
```

Comment:
This is not how it's done. Please see my comment about how to do it using the event mix in

Reply:
hum, I'm not sure to understand what I have to do :/

Reply:
https://github.com/home-assistant/home-assistant-polymer/pull/1146#discussion_r194765731